dynamic-relations/src/main/java/at/drm/service/DynamicRelationsPrintService.java

```diff
@@ -0,0 +1,127 @@
+package at.drm.service;
+
+import at.drm.exception.NoRelationDaoFoundException;
+import at.drm.model.RelationIdentity;
+import at.drm.model.RelationLink;
+import jakarta.annotation.PostConstruct;
+import jakarta.persistence.EntityManager;
+import java.lang.reflect.Modifier;
+import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.type.filter.AssignableTypeFilter;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class DynamicRelationsPrintService {
+
+    private final EntityManager entityManager;
+    private final RelationService relationService;
+
+    private Map<String, Class<RelationIdentity>> classBySimpleName;
+
+    public void printMappedRelations(RelationIdentity relationIdentity) {
+        TreeNodeRelationIdentity root = new TreeNodeRelationIdentity(relationIdentity, new LinkedList<>());
+        createTree(root);
+        printTree(root);
+    }
+
+    private void printTree(TreeNodeRelationIdentity root) {
+        Deque<TreeNodeRelationIdentity> stack = new ArrayDeque<>();
```

Comment:
I think you dont need the sepearte node by level map.

Could look like this and should print the same result:

```
Deque<Map.Entry<TreeNodeRelationIdentity, Integer>> stack = new ArrayDeque<>();
stack.push(Map.entry(root, 0));

StringBuilder result = new StringBuilder("\n");

while (!stack.isEmpty()) {
    var entry = stack.pop();
    var node = entry.getKey();
    int level = entry.getValue();

    result.append(" ".repeat(level))
          .append(node.object().getType())
          .append("\n");

    for (var child : node.childObjects()) {
        stack.push(Map.entry(child, level + 1));
    }
}

log.info(result.toString());
```